src/include/duckdb/common/cycle_counter.hpp

```diff
@@ -10,32 +10,72 @@
 
 #include "duckdb/common/helper.hpp"
 #include "duckdb/common/chrono.hpp"
+#include "duckdb/common/random_engine.hpp"
 
 namespace duckdb {
 
-//! The cycle counter can be used to measure elapsed cycles
+//! The cycle counter can be used to measure elapsed cycles for a function, expression and ...
+//! Optimized by sampling mechanism. Once per 100 times.
+//! //Todo Can be optimized further by calling RDTSC once per sample
 class CycleCounter {
 public:
-	//! Starts the timer
-	void Start() {
-		finished = false;
-		start = Tick();
+	CycleCounter() : random(0) {
```

Comment:
The random engine always uses the same seed. Is that intended?

Reply:
This seems like a premature optimization with potential of causing issues. How does delegating this tuple counter interact with e.g. CASE expressions or conjunctions, that can cause the execution of entire branches of expressions to be skipped?

Reply:
Hmm, fair. I have disabled this optimization.